### PR TITLE
Gets rid of all the direct /atom/movable subtypes [MDBIGNORE]

### PR DESCRIPTION
**Pull request by Cyberboss**
image

To quote the DM reference verbatim: `You should not create instances of /atom/movable but should use /obj and /mob for actual objects. This object definition exists solely to define variables and procedures related to motion.`

Changes all exist `/obj` overrides (aside from a couple lines in obj.dm) to `/obj/basic`.
Changes all direct `/atom/movable` subtypes to be subtypes of `/obj`

Regexes used:
`(/|\(|,| )obj/([a-zA-Z0-9_]+)(/|\()` -> `$1obj/basic/$2$3`
`(/|\(|,| )atom/movable/((?!proc)[a-zA-Z0-9_]+)(/|\()` => `$1obj/$2$3`
30 or so build errors needed manual fixing (couple typedefs, weird arguments, etc...)

The only reason we override `/atom/movable` subtypes instead of `/obj` subtypes is because we've polluted `/obj` so much we can't use it for anything else. This corrects that by moving all our existing `/obj` procs and vars to `/obj/basic`.

Now we get stuff like `/obj/screen` and `/obj/virtualspeaker` back with no overhead.

@tgstation/commit-access

### CODE REVIEW

**Review by Timberpoes — APPROVED** [review]

File: code/game/objects/obj.dm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/obj
+	uses_integrity = FALSE
+	animate_movement = SLIDE_STEPS
+	speech_span = SPAN_ROBOT
+
+	vis_flags = VIS_INHERIT_PLANE // when this be added to vis_contents of something it inherit something.plane, important for visualisation of obj in openspace.
```
Comment (Cyberboss) [inline code/game/objects/obj.dm:6]: Only non-regex change

**Comment by Cyberboss** [comment]
Fucked up somewhere

**Comment by Cyberboss** [comment]
VSC capped out when replacing the lines so it's still broken (specifically maps)
I'll leave it up for debate
will fix later

**Comment by AMonkeyThatCodes** [comment]
whats the overhead on atom/movable? this PR explains nothing and for once I'd agree with MSO if he closed a pr.

File: code/_onclick/adjacent.dm [hunk 2]
```diff
@@ -65,7 +65,7 @@
 	Adjacency (to anything else):
 	* Must be on a turf
 */
-/atom/movable/Adjacent(atom/neighbor, atom/target, atom/movable/mover)
+/obj/Adjacent(atom/neighbor, atom/target, atom/movable/mover)
```
Comment (Watermelon914) [inline code/_onclick/adjacent.dm:68]:
```suggestion
/atom/movable/Adjacent(atom/neighbor, atom/target, atom/movable/mover)
```

File: code/__HELPERS/icon_smoothing.dm [hunk 3]
```diff
@@ -143,7 +143,7 @@ DEFINE_BITFIELD(smoothing_junction, list(
 					. |= SOUTHEAST_JUNCTION
 
 
-/atom/movable/calculate_adjacencies()
+/obj/calculate_adjacencies()
```
Comment (Watermelon914) [inline code/__HELPERS/icon_smoothing.dm:146]:
```suggestion
/atom/movable/calculate_adjacencies()
```

File: code/_onclick/other_mobs.dm [hunk 4]
```diff
@@ -85,7 +85,7 @@
 	if(!can_interact(user))
 		. = min(., UI_UPDATE)
 
-/atom/movable/can_interact(mob/user)
+/obj/can_interact(mob/user)
```
Comment (Watermelon914) [inline code/_onclick/other_mobs.dm:88]:
```suggestion
/atom/movable/can_interact(mob/user)
```

File: code/game/atoms_movable.dm [hunk 5]
```diff
@@ -85,8 +85,11 @@
 	/// The degree of pressure protection that mobs in list/contents have from the external environment, between 0 and 1
 	var/contents_pressure_protection = 0
 
+	var/pressure_resistance = 10
+	var/last_high_pressure_movement_air_cycle = 0
 
-/atom/movable/Initialize(mapload)
+
+/obj/Initialize(mapload)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:92]:
```suggestion
/atom/movable/Initialize(mapload)
```

File: code/game/atoms_movable.dm [hunk 6]
```diff
@@ -108,7 +111,7 @@
 			AddComponent(/datum/component/overlay_lighting, is_directional = TRUE)
 
 
-/atom/movable/Destroy(force)
+/obj/Destroy(force)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:114]:
```suggestion
/atom/movable/Destroy(force)
```

File: code/game/atoms_movable.dm [hunk 7]
```diff
@@ -164,7 +167,7 @@
 			em_block = new(src, render_target)
 		return em_block
 
-/atom/movable/update_overlays()
+/obj/update_overlays()
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:170]:
```suggestion
/atom/movable/update_overlays()
```

File: code/game/atoms_movable.dm [hunk 8]
```diff
@@ -213,7 +216,7 @@
 	return T.zPassOut(src, direction, destination) && destination.zPassIn(src, direction, T)
 
 
-/atom/movable/vv_edit_var(var_name, var_value)
+/obj/vv_edit_var(var_name, var_value)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:219]:
```suggestion
/atom/movable/vv_edit_var(var_name, var_value)
```

File: code/game/atoms_movable.dm [hunk 9]
```diff
@@ -383,7 +386,7 @@
 // Here's where we rewrite how byond handles movement except slightly different
 // To be removed on step_ conversion
 // All this work to prevent a second bump
-/atom/movable/Move(atom/newloc, direction, glide_size_override = 0)
+/obj/Move(atom/newloc, direction, glide_size_override = 0)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:389]:
```suggestion
/atom/movable/Move(atom/newloc, direction, glide_size_override = 0)
```

File: code/game/atoms_movable.dm [hunk 10]
```diff
@@ -457,7 +460,7 @@
 
 ////////////////////////////////////////
 
-/atom/movable/Move(atom/newloc, direct, glide_size_override = 0)
+/obj/Move(atom/newloc, direct, glide_size_override = 0)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:463]:
```suggestion
/atom/movable/Move(atom/newloc, direct, glide_size_override = 0)
```

File: code/game/atoms_movable.dm [hunk 11]
```diff
@@ -587,14 +590,14 @@
 
 // Make sure you know what you're doing if you call this, this is intended to only be called by byond directly.
 // You probably want CanPass()
-/atom/movable/Cross(atom/movable/AM)
+/obj/Cross(atom/movable/AM)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:593]:
```suggestion
/atom/movable/Cross(atom/movable/AM)
```

File: code/game/atoms_movable.dm [hunk 12]
```diff
@@ -587,14 +590,14 @@
...
 	. = TRUE
 	SEND_SIGNAL(src, COMSIG_MOVABLE_CROSS, AM)
 	SEND_SIGNAL(AM, COMSIG_MOVABLE_CROSS_OVER, src)
 	return CanPass(AM, get_dir(src, AM))
 
 ///default byond proc that is deprecated for us in lieu of signals. do not call
-/atom/movable/Crossed(atom/movable/AM, oldloc)
+/obj/Crossed(atom/movable/AM, oldloc)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:600]:
```suggestion
/atom/movable/Crossed(atom/movable/AM, oldloc)
```

File: code/game/atoms_movable.dm [hunk 13]
```diff
@@ -618,7 +621,7 @@
  * replacement is [`/datum/element/connect_loc`] while hooking onto
  * [`COMSIG_ATOM_EXIT`].
  */
-/atom/movable/Uncross()
+/obj/Uncross()
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:624]:
```suggestion
/atom/movable/Uncross()
```

File: code/game/atoms_movable.dm [hunk 14]
```diff
@@ -628,11 +631,11 @@
  * this is wasteful since the vast majority of objects do not use Uncrossed
  * use connect_loc to register to COMSIG_ATOM_EXITED instead
  */
-/atom/movable/Uncrossed(atom/movable/AM)
+/obj/Uncrossed(atom/movable/AM)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:634]:
```suggestion
/atom/movable/Uncrossed(atom/movable/AM)
```

File: code/game/atoms_movable.dm [hunk 15]
```diff
@@ -628,11 +631,11 @@
...
 	SHOULD_NOT_OVERRIDE(TRUE)
-	CRASH("/atom/movable/Uncrossed() was called")
+	CRASH("/obj/Uncrossed() was called")
 
-/atom/movable/Bump(atom/A)
+/obj/Bump(atom/A)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:638]:
```suggestion
/atom/movable/Bump(atom/A)
```

File: code/game/atoms_movable.dm [hunk 16]
```diff
@@ -628,11 +631,11 @@
  * this is wasteful since the vast majority of objects do not use Uncrossed
  * use connect_loc to register to COMSIG_ATOM_EXITED instead
  */
-/atom/movable/Uncrossed(atom/movable/AM)
+/obj/Uncrossed(atom/movable/AM)
 	SHOULD_NOT_OVERRIDE(TRUE)
-	CRASH("/atom/movable/Uncrossed() was called")
+	CRASH("/obj/Uncrossed() was called")
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:636]:
```suggestion
	CRASH("/atom/movable/Uncrossed() was called")
```

File: code/game/atoms_movable.dm [hunk 17]
```diff
@@ -644,7 +647,7 @@
 			return
 	A.Bumped(src)
 
-/atom/movable/Exited(atom/movable/gone, direction)
+/obj/Exited(atom/movable/gone, direction)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:650]:
```suggestion
/atom/movable/Exited(atom/movable/gone, direction)
```

File: code/game/atoms_movable.dm [hunk 18]
```diff
@@ -653,7 +656,7 @@
 			for(var/atom/movable/location as anything in nested_locs)
 				LAZYREMOVEASSOC(location.important_recursive_contents, channel, gone.important_recursive_contents[channel])
 
-/atom/movable/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+/obj/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:659]:
```suggestion
/atom/movable/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
```

File: code/game/atoms_movable.dm [hunk 19]
```diff
@@ -816,7 +819,7 @@
 	if(!(impact_signal && (impact_signal & COMPONENT_MOVABLE_IMPACT_NEVERMIND))) // in case a signal interceptor broke or deleted the thing before we could process our hit
 		return hit_atom.hitby(src, throwingdatum=throwingdatum, hitpush=hitpush)
 
-/atom/movable/hitby(atom/movable/AM, skipcatch, hitpush = TRUE, blocked, datum/thrownthing/throwingdatum)
+/obj/hitby(atom/movable/AM, skipcatch, hitpush = TRUE, blocked, datum/thrownthing/throwingdatum)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:822]:
```suggestion
/atom/movable/hitby(atom/movable/AM, skipcatch, hitpush = TRUE, blocked, datum/thrownthing/throwingdatum)
```

File: code/game/atoms_movable.dm [hunk 20]
```diff
@@ -939,7 +942,7 @@
 /atom/movable/proc/move_crushed(atom/movable/pusher, force = MOVE_FORCE_DEFAULT, direction)
 	return FALSE
 
-/atom/movable/CanAllowThrough(atom/movable/mover, border_dir)
+/obj/CanAllowThrough(atom/movable/mover, border_dir)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:945]:
```suggestion
/atom/movable/CanAllowThrough(atom/movable/mover, border_dir)
```

File: code/game/atoms_movable.dm [hunk 21]
```diff
@@ -1049,7 +1052,7 @@
 	animate(time = 1)
 	animate(alpha = 0, time = 3, easing = CIRCULAR_EASING|EASE_OUT)
 
-/atom/movable/vv_get_dropdown()
+/obj/vv_get_dropdown()
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:1055]:
```suggestion
/atom/movable/vv_get_dropdown()
```

File: code/game/atoms_movable.dm [hunk 22]
```diff
@@ -1194,12 +1197,12 @@
 /atom/movable/proc/deadchat_plays(mode = ANARCHY_MODE, cooldown = 12 SECONDS)
 	return AddComponent(/datum/component/deadchat_control/cardinal_movement, mode, list(), cooldown)
 
-/atom/movable/vv_get_dropdown()
+/obj/vv_get_dropdown()
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:1200]:
```suggestion
/atom/movable/vv_get_dropdown()
```

File: code/game/atoms_movable.dm [hunk 23]
```diff
@@ -1194,12 +1197,12 @@
...
 	. = ..()
 	VV_DROPDOWN_OPTION(VV_HK_DEADCHAT_PLAYS, "Start/Stop Deadchat Plays")
 	VV_DROPDOWN_OPTION(VV_HK_ADD_FANTASY_AFFIX, "Add Fantasy Affix")
 
-/atom/movable/vv_do_topic(list/href_list)
+/obj/vv_do_topic(list/href_list)
```
Comment (Watermelon914) [inline code/game/atoms_movable.dm:1205]:
```suggestion
/atom/movable/vv_do_topic(list/href_list)
```

File: code/game/objects/buckling.dm [hunk 24]
```diff
@@ -13,7 +13,7 @@
 	var/buckle_prevents_pull = FALSE
 
 //Interaction
-/atom/movable/attack_hand(mob/living/user, list/modifiers)
+/obj/attack_hand(mob/living/user, list/modifiers)
```
Comment (Watermelon914) [inline code/game/objects/buckling.dm:16]:
```suggestion
/atom/movable/attack_hand(mob/living/user, list/modifiers)
```

File: code/game/objects/buckling.dm [hunk 25]
```diff
@@ -26,7 +26,7 @@
 			if(user_unbuckle_mob(buckled_mobs[1],user))
 				return TRUE
 
-/atom/movable/attackby(obj/item/W, mob/user, params)
+/obj/attackby(obj/item/W, mob/user, params)
```
Comment (Watermelon914) [inline code/game/objects/buckling.dm:29]:
```suggestion
/atom/movable/attackby(obj/item/W, mob/user, params)
```

File: code/game/objects/buckling.dm [hunk 26]
```diff
@@ -39,7 +39,7 @@
 	return mouse_buckle_handling(carried_mob, user)
 
 //literally just the above extension of attack_hand(), but for silicons instead (with an adjacency check, since attack_robot() being called doesn't mean that you're adjacent to something)
-/atom/movable/attack_robot(mob/living/user)
+/obj/attack_robot(mob/living/user)
```
Comment (Watermelon914) [inline code/game/objects/buckling.dm:42]:
```suggestion
/atom/movable/attack_robot(mob/living/user)
```

File: code/game/objects/buckling.dm [hunk 27]
```diff
@@ -50,7 +50,7 @@
 		else
 			return user_unbuckle_mob(buckled_mobs[1], user)
 
-/atom/movable/MouseDrop_T(mob/living/M, mob/living/user)
+/obj/MouseDrop_T(mob/living/M, mob/living/user)
```
Comment (Watermelon914) [inline code/game/objects/buckling.dm:53]:
```suggestion
/atom/movable/MouseDrop_T(mob/living/M, mob/living/user)
```

File: code/modules/keybindings/bindings_atom.dm [hunk 28]
```diff
@@ -1,7 +1,7 @@
 // You might be wondering why this isn't client level. If focus is null, we don't want you to move.
 // Only way to do that is to tie the behavior into the focus's keyLoop().
 
-/atom/movable/keyLoop(client/user)
+/obj/keyLoop(client/user)
```
Comment (Watermelon914) [inline code/modules/keybindings/bindings_atom.dm:4]:
```suggestion
/atom/movable/keyLoop(client/user)
```

File: code/modules/lighting/lighting_atom.dm [hunk 29]
```diff
@@ -63,7 +63,7 @@
 	opacity = new_opacity
 
 
-/atom/movable/set_opacity(new_opacity)
+/obj/set_opacity(new_opacity)
```
Comment (Watermelon914) [inline code/modules/lighting/lighting_atom.dm:66]:
```suggestion
/atom/movable/set_opacity(new_opacity)
```

File: code/modules/lighting/lighting_atom.dm [hunk 30]
```diff
@@ -81,7 +81,7 @@
 	recalculate_directional_opacity()
 
 
-/atom/movable/Moved(atom/OldLoc, Dir)
+/obj/Moved(atom/OldLoc, Dir)
```
Comment (Watermelon914) [inline code/modules/lighting/lighting_atom.dm:84]:
```suggestion
/atom/movable/Moved(atom/OldLoc, Dir)
```

File: code/modules/shuttle/shuttle_rotate.dm [hunk 31]
```diff
@@ -27,7 +27,7 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 
 /************************************Base /atom/movable proc************************************/
 
-/atom/movable/shuttleRotate(rotation, params)
+/obj/shuttleRotate(rotation, params)
```
Comment (Watermelon914) [inline code/modules/shuttle/shuttle_rotate.dm:30]:
```suggestion
/atom/movable/shuttleRotate(rotation, params)
```

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Comment by Gandalf2k15** [comment]
why

**Comment by Watermelon914** [comment]
here is why
https://www.byond.com/docs/ref/#/atom/movable

**Comment by AMonkeyThatCodes** [comment]
byond ref also tells us to use sleep and spawn. Can some more info just be put into the pr body??

**Comment by ZephyrTFA** [comment]
This seems unneeded. If you unironically trusted what the byond doc tells you is the best practice, you'd also be using sleep, spawn, and the ':' path operator

**Comment by nemvar** [comment]
> The only reason we override `/atom/movable` subtypes instead of `/obj` subtypes is because we've polluted `/obj` so much we can't use it for anything else. This corrects that by moving all our existing `/obj` procs and vars to `/obj/basic`.

facts

**Comment by Watermelon914** [comment]
I have checked both the sleep and the spawn reference pages and they do not say you should be using them in place of timers.
The byond ref explicitly states and says that you shouldn't be creating /atom/movable instances.

Also, doing research will tell you that the byond ref does not recommend you to use `:`
image

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 32]
```diff
@@ -339,9 +339,6 @@
 		if (!M.anchored && !M.pulledby && M.last_high_pressure_movement_air_cycle < SSair.times_fired)
 			M.experience_pressure_difference(pressure_difference, pressure_direction)
 
-/atom/movable/var/pressure_resistance = 10
-/atom/movable/var/last_high_pressure_movement_air_cycle = 0
-
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:344]: I'm pretty sure this was here in an attempt to keep /atmospherics modular to some extent, similar to the turf vars at the top of the file. Should it stay with uh, less absolute pathing?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
This regex makes me nervous

**Comment by AnturK** [comment]
Elaborate shitpost

**Comment by Mothblocks** [comment]
/obj/basic oml

**Comment by LemonInTheDark** [comment]
Should be called /obj/simple if you ask me

**Comment by optimumtact** [comment]
does this have a provable benefit?

Otherwise it's churn for OCD sake and is way too big for the result

**Comment by Kylerace** [comment]
as far as im aware, /obj has no inbuilt differences from /atom/movable. i dont see the point in this. i like the convention of having /atom/movable more or less be abstract types and /obj be physical types even if we are nowhere near adopting that as a standard and a lot of old code not doing so. but in any case i seriously doubt there is any mechanical reason to use /obj over /atom/movable, they have the exact same vars, and the exact same procs. not to mention this is a mass replace of the father typepath of mobs and obj into just obj which makes me nervous.

**Comment by Kylerace** [comment]
also if we're including our own addons to both: /atom/movable has significantly more procs than /obj. by my count: there are 23 vars on /obj and 79 procs, and 73 vars on /atom/movable and 141 procs. i dont see how we are polluting /obj

**Comment by Watermelon914** [comment]
Here is what lummox has to say about subtyping off of /atom/movable. The worst part, if he ever does it, is probably the last point he made.
image

**Comment by Kylerace** [comment]
this is still a LOT of churn to go through for a distant possibility. if it causes bugs with 51x then we dont update until its fixed or we go through this then.

**Comment by Cyberboss** [comment]
Don't care anymore

**Pull request closed without merging**